### ISSUE TITLE
[SR-2024] REPL doesn't show warning/error when `try` is redundant or required

### ISSUE THREAD

**Issue by benrimmington** [swiftlang/llvm-project#4517]
|                  |                 |
   |------------------|-----------------|
   |Previous ID       | SR-2024      |
   |Radar             | None         |
   |Original Reporter | @benrimmington      |
   |Type              | Bug    |

<details>
  <summary>Environment</summary>

Apple Swift version 4.2 (swiftlang-1000.11.37.1 clang-1000.11.45.1)

</details>

   <details>
  <summary>Additional Detail from JIRA</summary>

   |                  |                 |
   |------------------|-----------------|
   |Votes             | 0         |
   |Component/s       | LLDB for Swift    |
   |Labels            | Bug        |
   |Assignee          | None      |
   |Priority          | Medium      |

   

   md5: e14913509b81209ac40e8596206668dc

  </details>

**Issue Description:**

Xcode playground shows the expected warning/error:

``` java
func test1() {
    do {
        try print("foo") // ⚠️ No calls to throwing functions occur within 'try' expression
    } catch {            // ⚠️ 'catch' block is unreachable because no errors are thrown in 'do' block
        fatalError()
    }
}

func test2() throws {
    try print("foo") // ⚠️ No calls to throwing functions occur within 'try' expression
}

try test1() // ⚠️ No calls to throwing functions occur within 'try' expression

test2() // ⛔️ Call can throw but is not marked with 'try'
```

*REPL doesn't show any warning/error for the above code.*

<https://forums.swift.org/t/discussion-try-for-function-that-does-not-throw/3328>

**Comment by Dante-Broggi** [comment]
Is this resolved, or no longer valid? If either, this should be closed.

**Comment by benrimmington** [comment]
Tested with Swift 5.3-dev REPL (master branch). Still no warnings or errors.
